src/audio/host.c

```diff
@@ -481,6 +481,8 @@ static void host_buffer_cb(void *data, uint32_t bytes)
 
 	tracev_host("host_buffer_cb(), copy_bytes = 0x%x", copy_bytes);
 
+	/* mark buffer as busy until DMA transfer is done */
+	hd->dma_buffer->free = 0;
```

Comment:
Marcin, this is not good, force to set buffer->free here will break many use scenarios, we produced some data(not be full always) for the buffer, then call buffer->cb(), and initiate another dma copy().

We should always refresh buffer->free and buffer->avail at buffer produce()/consume() at runtime, don't change it manually unless you are at new()/prepare()/reset().